### ISSUE TITLE
Version v2.0.0 not usable

### ISSUE THREAD

**Issue by thomaspoignant** [nikunjy/rules#29]
Hello,

I was trying to use version `v2.0.0` of the lib but it is not possible to use it for now.

On a clean project when I do `go get github.com/nikunjy/rules` it downloads the version `v1.1.0`.
```shell
go: added github.com/antlr/antlr4/runtime/Go/antlr v0.0.0-20220415221154-79c36419192d
go: added github.com/blang/semver v3.5.1+incompatible
go: added github.com/nikunjy/rules v1.1.0
```

And when I try to force the go get of the version `v2.0.0` I have an error:
```shell
> go get github.com/nikunjy/rules@v2.0.0
go: github.com/nikunjy/rules@v2.0.0: invalid version: module contains a go.mod file, so module path must match major version ("github.com/nikunjy/rules/v2")
```

After reading here and there it seems that the issue is caused because your version `v2` is not following the GO convention.
Look at this [article](https://golang.cafe/blog/how-to-upgrade-to-a-major-version-in-go.html) to know more.

**Comment by thomaspoignant** [comment]
I just saw there is a PR link to that #28

**Comment by nikunjy** [comment]
You should be able to upgrade now @thomaspoignant  
v1.2.0 is released

**Comment by nikunjy** [comment]
I released a v1.2.0 so it should work now. I don't think I needed v2.X to begin with anyway 
Sent a PR your way @thomaspoignant  https://github.com/thomaspoignant/go-feature-flag/pull/405

**Comment by nikunjy** [comment]
Closing this issue as it seems like we are unblocked. Per this comment https://github.com/thomaspoignant/go-feature-flag/pull/405#issuecomment-1314973034
